sklearn/utils/validation.py

```diff
@@ -155,6 +156,34 @@ def _shape_repr(shape):
     return "(%s)" % joined
 
 
+def check_memory(memory):
+    """Check that the memory is an instance of joblib.Memory.
+
+    Raises a ValueError if the passed object does not have a
+    cache attribute.
+
+    Parameters
+    ----------
+    memory : string or None or Memory instance.
+        Any object with a cache method will be accepted as a Memory instance.
+
+    Returns
+    -------
+    memory : Memory
+        Validated memory instance.
+    """
+
+    if memory is None:
+        memory = Memory(cachedir=None, verbose=0)
+    elif isinstance(memory, six.string_types):
+        memory = Memory(cachedir=memory, verbose=0)
+    elif not hasattr(memory, 'cache'):
+            raise ValueError("'memory' is not a string or a Memory instance "
```

Comment:
Can you please give an example? I do not understand what to print in case `memory` is a class.

Reply:
@glemaitre Can you please give an example of how to achieve this?

Reply:
I think that what @lesteve is mentioning is to `print(memory)` instead of `print(type(memory))`
In short put the value of memory instead of the type of memory.